src/memory_manager.rs

```diff
@@ -419,6 +561,43 @@ impl<M: Memory> MemoryManagerInner<M> {
         // Ceiling division.
         (num_pages + self.bucket_size_in_pages as u64 - 1) / self.bucket_size_in_pages as u64
     }
+
+    fn free(&mut self, id: MemoryId) {
+        self.memory_sizes_in_pages[id.0 as usize] = 0;
+        let buckets = self.memory_buckets.remove(&id);
+        if let Some(vec_buckets) = buckets {
+            for bucket in vec_buckets {
+                self.unallocated_buckets.push_front(bucket);
+                self.allocated_buckets -= 1;
+            }
+        }
+        // Update the header.
+        self.save_header();
+
+        // Write in stable store that no bucket belongs to the memory with the provided `id`.
+        write(
+            &self.memory,
+            bucket_allocations_address(BucketId(0)).get(),
```

Comment:
I think the use `bucket_allocations_address(BucketId(0))` here and elsewhere is misleading. The function `bucket_allocations_address` is specific to V1, where it returns the address where we store the allocation of the given bucket. That's not how they're being used in this code.

I think what your code truly cares about is the offset where the bucket indexes are stored. So instead of using `bucket_allocations_address(BucketId(0))`, we instead can define a constant, say `BUCKET_INDEXES_OFFSET` with that value.

Reply:
I don't understand what you mean "consistent with nonconstant values", but it definitely can be a constant. I'll do it.